src/server/pfs/server/driver_v2.go

```diff
@@ -478,6 +473,53 @@ func (d *driverV2) globFileV2(pachClient *client.APIClient, commit *pfs.Commit,
 	})
 }
 
-func compactedCommitPath(commit *pfs.Commit) string {
-	return path.Join(commitKey(commit), fileset.Compacted)
+func (d *driverV2) inspectFile(pachClient *client.APIClient, file *pfs.File) (ret *pfs.FileInfoV2, retErr error) {
+	ctx := pachClient.Ctx()
+	_, err := d.inspectCommit(pachClient, file.Commit, pfs.CommitState_FINISHED)
+	if err != nil {
+		return nil, err
+	}
+	p := cleanPath(file.Path)
+	s := NewSource(file.Commit, true, func() fileset.FileSource {
+		x := d.storage.NewSource(ctx, compactedCommitPath(file.Commit), index.WithPrefix(p))
+		x = fileset.NewIndexResolver(x)
+		x = fileset.NewIndexFilter(x, func(idx *index.Index) bool {
+			return idx.Path == p || strings.HasPrefix(idx.Path, p+"/")
+		})
+		return x
+	})
+	s = InsertNotFound(s, file)
```

Comment:
This seems like it would be better as a helper function, but naming it seems a bit weird (something like `RequireFileIterate` or `RequireNonEmptyIterate` might make sense?). Right now it just checks if nothing was iterated over, then returns the passed in file in an error if so. I think we need to pick either handling just an empty iteration or a particular file and checking for the file in the iteration (we don't check that the particular file was the one we iterated over).

Reply:
Thoughts on this?